fury/actor.py

```diff
@@ -1270,3 +1272,145 @@ def marker(
 
     obj = create_point(geo, mat)
     return obj
+
+
+def slicer(
+    data,
+    *,
+    value_range=None,
+    opacity=1.0,
+    interpolation="linear",
+    visibility=(True, True, True),
+    initial_slices=None,
+):
+    """Visualize a 3D volume data as a slice.
+
+    Parameters
+    ----------
+    data : ndarray, shape (X, Y, Z) or (X, Y, Z, 3)
+        The 3D volume data to be sliced.
+    value_range : tuple, optional
+        The minimum and maximum values for the color mapping.
+        If None, the range is determined from the data.
+    opacity : float, optional
+        The opacity of the slice. Takes values from 0 (fully transparent) to 1 (opaque).
+    interpolation : str, optional
+        The interpolation method for the slice. Options are 'linear' and 'nearest'.
+    visibility : tuple, optional
+        A tuple of three boolean values indicating the visibility of the slices
+        in the x, y, and z dimensions, respectively.
+    initial_slices : tuple, optional
+        A tuple of three initial slice positions in the x, y, and z dimensions,
+        respectively. If None, the slices are initialized to the middle of the volume.
+
+    Returns
+    -------
+    slicer_actor : Group
+        An actor containing the generated slice with the specified properties.
+    """
+
+    if value_range is None:
+        value_range = (np.min(data), np.max(data))
+
+    if visibility is None:
+        visibility = (True, True, True)
+
+    if data.ndim < 3 or data.ndim > 4:
+        raise ValueError(
+            "Input data must be 3-dimensional or "
+            "4-dimensional with last dimension of size 3."
+        )
+    elif data.ndim == 4 and data.shape[-1] != 3:
+        raise ValueError("Last dimension must be of size 3.")
+
+    opacity = validate_opacity(opacity)
+    data = data.astype(np.float32)
+
+    data = np.swapaxes(data, 0, 2)
+
+    data_shape = data.shape
+    if initial_slices is None:
+        initial_slices = (
+            data_shape[2] // 2,
+            data_shape[1] // 2,
+            data_shape[0] // 2,
+        )
+
+    texture = Texture(data, dim=3)
+
+    slices = []
+    for dim in [0, 1, 2]:  # XYZ
+        abcd = [0, 0, 0, 0]
+        abcd[dim] = -1
+        abcd[-1] = data_shape[2 - dim] // 2
+        mat = VolumeSliceMaterial(
+            abcd,
+            clim=value_range,
+            interpolation=interpolation,
+            opacity=opacity,
+            pick_write=True,
+        )
+        geo = Geometry(grid=texture)
+        plane = Volume(geo, mat)
+        slices.append(plane)
+
+    class Slicer(Group):
```

Comment:
I do not see the benefit of this class. 

Please create utils function that can be applied to a group. 

for example. `set_visibility` is not specific to slicer. it could be a simple function `set_group_visibility` which will work on any type of group later on. Same logic for all function

Reply:
The problem is the slice functions are specific to planes and not all groups will have planes in them they can have something different. For example, gfx.Scene is also a subclass of Group.